aws/resource_instance_profile.go

```diff
@@ -182,6 +201,18 @@ func ValidInstanceProfile(v any, c cty.Path) diag.Diagnostics {
 			},
 		}
 	}
+	var arnType string
+	switch c[0].(cty.GetAttrStep).Name {
+	case "instance_profile_arn":
+		arnType = "instance-profile"
+	case "instance_profile_id":
+		arnType = "instance-profile"
+	case "iam_role_arn":
+		arnType = "role"
+	}
```

Comment:
should we have `default` here?

Reply:
Yeah, Otherwise the check down the line is against an empty string and whatever failure message comes out of it would be quite useless.